zerver/tests/test_message_send.py

```diff
@@ -1317,6 +1317,108 @@ def test_message_bot_mentions(self) -> None:
         self.assertEqual(user_message.message.content, content)
         self.assertTrue(user_message.flags.mentioned)
 
+    def test_wildcard_mention_restrictions(self) -> None:
+        cordelia = self.example_user("cordelia")
+        iago = self.example_user("iago")
+        self.login_user(cordelia)
+
+        stream_name = "test_stream"
+        self.subscribe(cordelia, stream_name)
+        self.subscribe(iago, stream_name)
+
+        # When wildcard_mention_policy is set to
+        # WILDCARD_MENTION_SHOW_WARNING and a message with wildcard
+        # mention is sent through API, we simply send the
+        # message. Warning is only shown when message is sent through
+        # UI.
+        content = "@**all** test wildcard mention"
+        result = self.client_post(
+            "/json/messages",
+            {
+                "type": "stream",
+                "to": "test_stream",
+                "client": "test suite",
+                "content": content,
+                "topic": "Test topic",
+            },
+        )
+        self.assert_json_success(result)
+        last_message = self.get_last_message()
+        self.assertEqual(last_message.content, content)
```

Comment:
We should extract this block as a helper function; probably `self.send_stream_message()` to save some duplicated code, and maybe a small wrapper for this specific test.  I'm not sure we need to do the `last_message` / `content` verification.

Reply:
(And we should test all 6 settings values).